ui/pages/settings/networks-tab/networks-form/rpc-url-editor.tsx

```diff
@@ -0,0 +1,139 @@
+import React, { useRef, useState } from 'react';
+import classnames from 'classnames';
+import {
+  Box,
+  ButtonIcon,
+  ButtonIconSize,
+  ButtonLink,
+  Icon,
+  IconName,
+  IconSize,
+  Popover,
+  PopoverPosition,
+  Text,
+} from '../../../../components/component-library';
+import {
+  AlignItems,
+  BackgroundColor,
+  BorderColor,
+  BorderRadius,
+  Display,
+  IconColor,
+  JustifyContent,
+  TextColor,
+  TextVariant,
+} from '../../../../helpers/constants/design-system';
+import { useI18nContext } from '../../../../hooks/useI18nContext';
+
+export const RpcUrlEditor = ({ currentRpcUrl }: { currentRpcUrl: string }) => {
+  // TODO: real endpoints
+  const dummyRpcUrls = [
+    currentRpcUrl,
+    'https://dummy-mainnet.public.blastapi.io',
+    'https://dummy.io/v3/blockchain/node/dummy',
+  ];
+
+  const t = useI18nContext();
+  const rpcDropdown = useRef(null);
+  const [isOpen, setIsOpen] = useState(false);
+  const [currentRpcEndpoint, setCurrentRpcEndpoint] = useState(currentRpcUrl);
+
+  return (
+    <>
+      <Text
+        className="networks-tab__rpc-header"
+        marginTop={1}
+        marginBottom={1}
+        variant={TextVariant.bodySmBold}
+      >
+        {t('defaultRpcUrl')}
+      </Text>
+      <Box
```

Comment:
Can this be a `button`?  Use the `as="button"` prop?

Reply:
Ah this does need to be accessible.  Doing that messed up a lot of styling, so it'd be easier to make the `Icon` a `ButtonIcon`